cmake/dts.cmake

```diff
@@ -163,7 +163,7 @@ if(SUPPORTS_DTS)
 
   set(CMD_NEW_EXTRACT ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/dts/gen_defines.py
   --dts ${BOARD}.dts.pre.tmp
-  --bindings-dir ${DTS_ROOT_BINDINGS}
+  --bindings-dir "${DTS_ROOT_BINDINGS}"
```

Comment:
Will this be a semicolon-separated list? The script doesn't know how to deal with that. Spaces will probably break too.

Reply:
Not even sure how the `argparse` Python module deals with `--some-flag` combined with `nargs="+"`. Feels like it could lead to collisions with other command-line arguments, if it means that all arguments after `--bindings-dir` should be taken as a path.

Reply:
Yes, cmake constructs it as a semicolon-separated list.  The quotes would also be necessary if path components included spaces, but `dts.cmake` blows chunks in that situation anyway.

It sounds like the scripts need to be updated to remove `nargs="+"` and require semi-colon separated entries for those arguments.

Reply:
I think `argparse` supports collecting arguments into a list. That might be the cleanest solution. Might have to pass multiple `bindings-dir` for it though.

Reply:
Kinda like how `-I<dir>` works in GCC. Might have to build up something like `--bindings-dir some/path --bindings-dir some/other/path` in CMake.

Reply:
Experimentation suggests that `argsparse` is using the semicolon separator.  Since the quotes were only added proactively to deal with paths that include spaces, and dts.cmake doesn't handle that case anyway, I just took those changes out.

Reply:
Not seeing it for this script:

```python
import argparse

parser = argparse.ArgumentParser()
parser.add_argument("--foo", nargs="+")

print(parser.parse_args())
```

Running

```sh
$ python3 args.py --foo 'bar;baz'
```
outputs

```
Namespace(foo=['hej;san'])
```

What we probably want is the [`append` action](https://docs.python.org/3/library/argparse.html#action).